store/pg_engine_test.go

```diff
@@ -47,3 +59,61 @@ func TestGetMigrationVersions(t *testing.T) {
 		require.Equal(t, test.want, migrateVersions)
 	}
 }
+
+var (
+	pgUser        = "test"
+	pgPort        = 6000
+	serverVersion = "server-version"
+	l             = zap.NewNop()
+)
+
+func TestMigrationCompatibility(t *testing.T) {
+	pgDir := t.TempDir()
+	pgInstance, err := postgres.Install(path.Join(pgDir, "resource"), path.Join(pgDir, "data"), pgUser)
+	require.NoError(t, err)
+	err = pgInstance.Start(pgPort, os.Stdout, os.Stderr)
+	require.NoError(t, err)
+	defer pgInstance.Stop(os.Stdout, os.Stderr)
+
+	ctx := context.Background()
+	connCfg := dbdriver.ConnectionConfig{
+		Username: pgUser,
+		Password: "",
+		Host:     common.GetPostgresSocketDir(),
+		Port:     fmt.Sprintf("%d", pgPort),
+	}
+	d, err := dbdriver.Open(
+		ctx,
+		dbdriver.Postgres,
+		dbdriver.DriverConfig{Logger: l},
+		connCfg,
+		dbdriver.ConnectionContext{},
+	)
+	require.NoError(t, err)
+
+	err = d.SetupMigrationIfNeeded(ctx)
+	require.NoError(t, err)
+
+	versions, err := getMigrationFileVersions()
+	require.NoError(t, err)
+
+	// Setup N databases with N versions in the migration directory with latest schema, and apply migrations till the latest version.
```

Comment:
Let's add the example to explain it. The sentence is still a bit hard to understand.

Reply:
Done.